SBAT.example.md

```diff
@@ -0,0 +1,224 @@
+SBAT: Current proposal
+================
+
+`.sbat` section
+-------------
+
+the `.sbat` section has the following fields:
+| field | meaning |
+|---|---|
+| component_name | the name we're comparing
+| component_generation | the generation number for the comparison
+| vendor_name | human readable vendor name
+| vendor_package_name | human readable package name
+| vendor_version | human readable package version (maybe machine parseable too, not specified here)
+| vendor_url | url to look stuff up, contact, whatever.
+
+`SBAT` EFI variable
+-----------------
+The SBAT EFI variable (`SBAT-605dab50-e046-4300-abb6-3dd810dd8b23`) is structured as a series of records:
+
+```
+struct sbat_entry {
+  uint16_t component_generation;
+  uint16_t component_name_size;
+  char component_name[];
+}
+```
+
+with the first record being our structure version number expressed as: `{ 0, 5, "sbat" }`
+
+Validation Rules
+-----------------
+- If an `SBAT` variable is set, binaries validated directly by shim (i.e. not with the API) get validated using SBAT
+- If a binary validated by the API *does* have a `.sbat` section, it also gets validating using SBAT
+- If a binary is enrolled in `db` by its hash rather than by certificate, the validation result is logged only, not enforced
+- When validating SBAT, any `component_name` that's in both `SBAT` and `.sbat` gets compared
+- `component_generation` in `.sbat` must be >= the highest `component_generation` of an identical `component_name` in `SBAT`
+- that version comparison includes the `"sbat"` entry
+
+Example universe starting point
+-------------------------------
+For grub, a build from a fresh checkout of upstream might have the following in
+`.sbat`:
+```
+sbat,0,UEFI shim,0,https://github.com/rhboot/shim/blob/main/SBAT.md
```

Comment:
This entry is missing a `vendor_package_name`. Should it be `sbat` again or ?

Reply:
yep.  fixed in latest push.